Tests/test_file_msp.py

```diff
@@ -24,14 +27,43 @@ def test_invalid_file(self):
         self.assertRaises(SyntaxError,
                           lambda: MspImagePlugin.MspImageFile(invalid_file))
 
-    def test_open(self):
+    def test_bad_checksum(self):
+        # Arrange
+        # This was created by forcing Pillow to save with checksum=0
+        bad_checksum = "Tests/images/hopper_bad_checksum.msp"
+
+        # Act / Assert
+        self.assertRaises(SyntaxError,
+                          lambda: MspImagePlugin.MspImageFile(bad_checksum))
+
+    def test_open_windows_v1(self):
         # Arrange
         # Act
         im = Image.open(TEST_FILE)
 
         # Assert
         self.assertEqual(im.size, (128, 128))
         self.assert_image_equal(im, hopper("1"), 4)
+        self.assertIsInstance(im, MspImagePlugin.MspImageFile)
+
+    @unittest.skipIf(not os.path.exists(EXTRA_DIR),
+                     "Extra image files not installed")
+    def test_open_windows_v2(self):
+        # Arrange
+        ImageFile.LOAD_TRUNCATED_IMAGES = True
```

Comment:
Ah. It's YAFRLE implementation. 

That sort of thing really sounds like an RLE decoding error in MspDecode.c, as RLE is prone to underrun or overrun if it's interpreted incorrectly. (e.g. https://github.com/python-pillow/Pillow/issues/2231). I'd call this probably broken unless I can prove otherwise.